app/decorators/case_transition_decorator.rb

```diff
@@ -24,6 +24,15 @@ def event
     state_machine.event_name(object.event)
   end
 
+  def event_desc
+    specific_key = "event.case/#{object.case.type_abbreviation.downcase}.#{object.event}"
+    specific_key_with_desc = "event.case/#{object.case.type_abbreviation.downcase}.#{object.event}__desc"
+    default_key = "event.#{object.event}"
+    default_key_with_desc = "event.#{object.event}__desc"
+    I18n.t(specific_key_with_desc, default: nil) || I18n.t(default_key_with_desc, default: nil) \
+    || I18n.t(specific_key, default: nil) || I18n.t(default_key, default: nil) || event
```

Comment:
I think this method is a tad unclear, and took me a while to work out what's going on. It looks like it could maybe do with being split into a few smaller methods?

I'd love to see a nicer way of structuring that massive line of "or"s and the default "nil" looks like these are hidden conditionals - is the idea we're checking for a translation for the first one then if it's nil we try the second and so on?

I think if you start moving things out into one-line, well-named private methods a nicer way forward might emerge

Also, maybe a `has_translation_for(key)` method something like this?
``` 
def has_translation_for(
  I18n.t(key, default: nil)
end
```
Or maybe not. Anyway, do you want to have another run at it?

Reply:
@Polsonby let me re-coding this line ( actually I had a feeling you will pick up this line)